src/point.c

```diff
@@ -263,3 +264,40 @@ spherepoint_equal(PG_FUNCTION_ARGS)
 
 	PG_RETURN_BOOL(spoint_eq(p1, p2));
 }
+
+static SPoint * spherepoint_from_vector3d(Vector3D v){
+	SPoint* p = (SPoint *) palloc(sizeof(SPoint));
+	p->lat = asin(v.z / sqrt(pow(v.x, 2) + pow(v.y, 2)  + pow(v.z, 2)));
+	p->lng = atan2(v.y, v.x);
+	return p;
+}
+
+Datum get_gravity_center(PG_FUNCTION_ARGS) {
+	SPoint * p;
+	SPoint current_point;
+	Vector3D	v;
+	Vector3D	res;
+	ArrayType *dots_vector = PG_GETARG_ARRAYTYPE_P(0);
+	int num_elements = ArrayGetNItems(ARR_NDIM(dots_vector), ARR_DIMS(dots_vector));
+	SPoint *array_data = (SPoint *) ARR_DATA_PTR(dots_vector);
+
+	for (int i = 0; i < num_elements; i++) {
+		current_point = array_data[i];
+		//elog(LOG, "%lf\n", current_point.lat);
```

Comment:
I would remove this line of code that has been commented out.

Reply:
I would also remote int i to the function beginning. Standard C doesn't allow to declare types such the way.